posthog/temporal/data_imports/pipelines/mongo/mongo.py

```diff
@@ -416,148 +337,51 @@ def mongo_source(
-                    "port": connection_params["port"] or 27017,
-                    "serverSelectionTimeoutMS": 5000,
-                }
-
-                if connection_params["user"] and connection_params["password"]:
-                    connection_kwargs.update(
-                        {
-                            "username": connection_params["user"],
-                            "password": connection_params["password"],
-                            "authSource": connection_params["auth_source"],
-                        }
-                    )
-
-                if connection_params["tls"]:
-                    connection_kwargs["tls"] = True
-
-                read_client = MongoClient(**connection_kwargs)
+    def get_rows() -> Iterator[dict[str, Any]]:
+        # New connection for data reading
+        read_client = _create_mongo_client(connection_string, connection_params)
 
         read_db = read_client[connection_params["database"]]
         read_collection = read_db[collection_name]
 
+        # TODO: update to pymongoarrow when pyarrow major version is bumped
         try:
-            cursor = read_collection.find(query)
-
-            if is_incremental and incremental_field:
-                cursor = cursor.sort(incremental_field, 1)  # ascending order
+            cursor = read_collection.find({})
 
-            batch = []
             for doc in cursor:
                 # Convert ObjectId to string and handle nested objects
                 processed_doc = {}
 
-                # First, initialize all schema fields with None
-                for field_name, _ in schema_info:
-                    processed_doc[field_name] = None
-
-                # Then populate with actual document values
+                # Process the document to handle ObjectIds and nested structures
                 for key, value in doc.items():
                     if isinstance(value, ObjectId):
                         processed_doc[key] = str(value)
                     elif isinstance(value, dict):
-                        processed_doc[key] = str(value)  # JSON as string
+                        # Keep nested objects as they are, but convert ObjectIds within them
+                        processed_doc[key] = _process_nested_object(value)
                     elif isinstance(value, list):
-                        processed_doc[key] = [str(item) if isinstance(item, ObjectId) else item for item in value]
+                        processed_doc[key] = [_process_nested_value(item) for item in value]
                     else:
                         processed_doc[key] = value
 
-                batch.append(processed_doc)
+                result = {
+                    "_id": str(doc["_id"]),
+                }
+                # extract incremental field from the document if it exists
+                if incremental_field:
+                    incremental_value = processed_doc.pop(incremental_field)
```

Comment:
Why are we popping the incremental field here? What if the user wants this to exist within `data` too? I'd say keep it on both the root and embedded in `data`